src/stavetie.ts

```diff
@@ -122,51 +160,66 @@ export class StaveTie extends Element {
     }
   }
 
-  renderText(first_x_px, last_x_px) {
+  renderText(first_x_px: number, last_x_px: number): void {
     if (!this.text) return;
+    const ctx = this.checkContext();
     let center_x = (first_x_px + last_x_px) / 2;
-    center_x -= this.context.measureText(this.text).width / 2;
-
-    this.context.save();
-    this.context.setFont(this.font.family, this.font.size, this.font.style);
-    this.context.fillText(
-      this.text,
-      center_x + this.render_options.text_shift_x,
-      (this.first_note || this.last_note).getStave().getYForTopText() - 1
-    );
-    this.context.restore();
+    center_x -= ctx.measureText(this.text).width / 2;
+    let stave: Stave;
+    const firstNoteStave = this.notes.first_note ? this.notes.first_note.getStave() : undefined;
+    const lastNoteStave = this.notes.last_note ? this.notes.last_note.getStave() : undefined;
+    if (firstNoteStave) {
+      stave = firstNoteStave;
+    } else if (lastNoteStave) {
+      stave = lastNoteStave;
+    } else {
+      throw new Vex.RERR('NoStave', 'Stave required');
+    }
```

Comment:
Seems like you can get the same effect with:

```typescript
let stave = this.notes.first_note?.getStave() ?? this.notes.last_note?.getStave();

...
ctx.fillText(..., stave!.getYForTopText() - 1)
...
```

Reply:
done

Reply:
done